pkg/controller/controller.go

```diff
@@ -270,14 +270,13 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		)
 		return 0, err
 	}
+	// If the node count is larger than the maximum, we want to attempt to scale down
 	if len(allNodes) > nodeGroup.Opts.MaxNodes {
```

Comment:
We should just get rid of this early safety check now that we're wanting to handle this edge case